### PR TITLE
Signed value support

### PR DESCRIPTION
**Pull request by 51systems**
Add support for SBYTE, SSHORT, SLONG, SRATIONAL as per the Tiff 6.0 spec
Add support for multiple BYTE

### CODE REVIEW

File: src/decoder/ifd.rs [hunk 1]
```diff
@@ -166,14 +201,51 @@ impl Entry {
...
             // at a different endianess of file/computer.
             (Type::BYTE, 1) => Ok(Unsigned(u32::from(self.offset[0]))),
+            (Type::BYTE, 2) => offset_to_bytes(2, self),
+            (Type::BYTE, 3) => offset_to_bytes(3, self),
+            (Type::BYTE, 4) => offset_to_bytes(4, self),
+            (Type::BYTE, n) => {
+                if n as usize > limits.decoding_buffer_size / mem::size_of::<Value>() {
+                    return Err(TiffError::LimitsExceeded);
+                }
+                let mut v = Vec::with_capacity(n as usize);
```
Comment (HeroicKatora) [inline src/decoder/ifd.rs]: Now would be a good opportunity to factor out these 4 lines, reused across many entry types, into a dedicated method.

Comment (51systems) [reply src/decoder/ifd.rs]: Done via closure

**Comment by birktj** [comment]
Looks good to me

**Comment by HeroicKatora** [comment]
Great work, thank you. You seem to be invested in `tiff`. Want to join `image-rs`? You can be invited to the `image-rs` collaborators team, which allows you access to this and the other repos if you want. See [the contribution policy](https://github.com/image-rs/organization/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/image-rs/organization/blob/master/CODE_OF_CONDUCT.md).

**Comment by 51systems** [comment]
That would be great! I'm still pretty new to both rust and image processing, but I'd like to contribute the dng encoder if it ever gets to an acceptable state.

**Pull request merged**